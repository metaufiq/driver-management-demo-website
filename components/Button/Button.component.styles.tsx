import styled from "styled-components";

export const StyledButton = styled.button`
  font-size: 0.75rem;
  margin: 1rem;
  padding: 1rem;
  border: 0;
`

export const PrimaryButton = styled(StyledButton)`
  background-color: #ff453b;
  color: white;
  border-radius: 0.125rem;
`

export const NavigationButton = styled(StyledButton)`
  background-color: transparent;
`