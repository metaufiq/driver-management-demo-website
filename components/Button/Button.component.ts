import { Button } from "./Button.component.styles";

export default Button