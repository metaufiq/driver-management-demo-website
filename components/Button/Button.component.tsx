import { Component } from "../../index.types";
import { Props } from "./Button.component.types";

const Button: Component<Props> = ({onClick, children, disabled})=>{
  return(
    <button 
      type="button"
      onClick={onClick}
      disabled={disabled}>
      {children}
    </button>
  )
}

export default Button;