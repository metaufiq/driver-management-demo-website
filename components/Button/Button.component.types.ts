export type Type = 'PRIMARY' | 'NAVIGATION';

export type Props = {
  textAlign?: 'start' | 'end';
  buttonType?: Type;
}