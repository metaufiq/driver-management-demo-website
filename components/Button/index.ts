import styled from "styled-components";
import { Props } from "./Button.component.types";

export const StyledButton = styled.button<Props>`
  font-size: 1rem;
  margin: 1rem;
  padding: 1rem;
  border: 0;
  text-align: ${({textAlign})=> textAlign ?? 'center'};
  display: flex;
  column-gap: 1rem;
  align-items: center;
  cursor: pointer;
`

export const PrimaryButton = styled(StyledButton)<Props>`
  background-color: #ff453b;
  color: white;
  border-radius: 0.125rem;
`

export const NavigationButton = styled(StyledButton)<Props>`
  font-size: 1rem;
  background-color: transparent;
`