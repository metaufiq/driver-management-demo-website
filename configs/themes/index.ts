const themes = {
  colors: {
    primary: '#ff453b'
  },
};

export { themes };