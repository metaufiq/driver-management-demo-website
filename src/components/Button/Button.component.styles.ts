import styled, { DefaultTheme, ThemedStyledProps } from "styled-components";
import { Props } from "./Button.component.types";

const _getBackgroundColor = ({buttonType, theme}:ThemedStyledProps<Props, DefaultTheme>)=>{
  switch (buttonType) {
    case "PRIMARY":
      return theme.colors.primary
    case "NAVIGATION":
      return 'transparent'
    default:
      return undefined
  }
}

const _getColor = ({buttonType}:ThemedStyledProps<Props, DefaultTheme>)=>{
  switch (buttonType) {
    case "PRIMARY":
      return 'white'
    default:
      return undefined
  }
}

export const Button = styled.button<Props>`
  font-size: 1rem;
  padding: 1rem;
  border: 0;
  text-align: ${(props)=> props.textAlign ?? 'center'};
  display: flex;
  column-gap: 1rem;
  align-items: center;
  background-color: ${_getBackgroundColor};
  color:${_getColor};
  cursor: pointer;
  :disabled{
    cursor: default;
  }
`