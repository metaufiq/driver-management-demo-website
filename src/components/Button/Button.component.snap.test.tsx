import { assertSnapshots } from "../../test";
import Button from ".";
import { Configs } from "../../test/index.types";
import { Props } from "./Button.component.types";
import { themes } from "../../configs/themes";

describe('StyledButtonSnapshot', () => {

  const configs:Configs<Props> = [
    {
      desc: 'should render normal button',
      useTheme: true,
      theme: themes
    },
    {
      desc: 'should render custom text align button',
      props: {textAlign:"start"},
      useTheme: true,
      theme: themes
    },
    {
      desc: 'should render normal primary button',
      props: {buttonType:'PRIMARY'},
      useTheme: true,
      theme: themes
    },
    {
      desc: 'should render normal navigation button',
      props: {buttonType:'NAVIGATION'},
      useTheme: true,
      theme: themes
    },
  ];

  assertSnapshots(Button, configs);
});
