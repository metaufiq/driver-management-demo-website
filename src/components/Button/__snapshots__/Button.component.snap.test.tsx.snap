// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StyledButtonSnapshot should render custom text align button 1`] = `
<button
  className="sc-bczRLJ dqWClK"
/>
`;

exports[`StyledButtonSnapshot should render normal button 1`] = `
<button
  className="sc-bczRLJ jMYSRp"
/>
`;

exports[`StyledButtonSnapshot should render normal navigation button 1`] = `
<button
  className="sc-bczRLJ hQcwZr"
/>
`;

exports[`StyledButtonSnapshot should render normal primary button 1`] = `
<button
  className="sc-bczRLJ iyjxpd"
/>
`;
